src/date-range-picker/calendar/__tests__/calendar-month.test.tsx

```diff
@@ -0,0 +1,654 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import * as React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import Mockdate from 'mockdate';
+
+import DateRangePicker, { DateRangePickerProps } from '../../../../lib/components/date-range-picker';
+import { NonCancelableEventHandler } from '../../../../lib/components/internal/events';
+import { KeyCode } from '../../../../lib/components/internal/keycode';
+import createWrapper from '../../../../lib/components/test-utils/dom';
+import DateRangePickerWrapper from '../../../../lib/components/test-utils/dom/date-range-picker';
+import { changeMode } from '../../__tests__/change-mode';
+import { i18nStrings } from '../../__tests__/i18n-strings';
+import { isValidRange } from '../../__tests__/is-valid-range';
+
+import testutilStyles from '../../../../lib/components/date-range-picker/test-classes/styles.selectors.js';
+import screenreaderOnlyStyles from '../../../../lib/components/internal/components/screenreader-only/styles.selectors.js';
+
+beforeEach(() => Mockdate.set(new Date('2020-10-20T12:30:20')));
+afterEach(() => Mockdate.reset());
+
+describe('Date range picker calendar with month granularity', () => {
+  const outsideId = 'outside';
+  const defaultProps: DateRangePickerProps & Pick<Required<DateRangePickerProps>, 'i18nStrings'> = {
```

Comment:
There is `SomeRequired` util, if you want: https://github.com/cloudscape-design/components/blob/372abbda82c19bd662bf9eac950092c699640320/src/internal/types.ts#L16-L18